### PR TITLE
bug/Incorrect pub key recovery

### PR DESCRIPTION
**Pull request by JGiter**
### Summary
|             |   |
|-------------|---|
| Description | Fixes public key related bugs |
| Jira issue  |   |

#### Type of request

- [x] Bug fix

### List of Fixes
- Fixed public key recovery algorithm
- Private key in claims package optional now, because it can't be calculated from Signer

### CODE REVIEW

File: packages/claims/src/claims/claims.ts [hunk 1]
```diff
@@ -54,10 +58,13 @@ export class Claims implements IClaims {
...
       return this.keys.publicKey;
     }
-    const hash = await ethers.utils.keccak256(await this.signer.getAddress());
-    const sig = await this.signer.signMessage(ethers.utils.arrayify(hash));
+    const hash = keccak256(await this.signer.getAddress());
+    const sig = await this.signer.signMessage(arrayify(hash));
+    const digest = arrayify(hashMessage(arrayify(hash)));
     // eslint-disable-next-line no-return-assign
-    return this.keys.publicKey = Promise.resolve(ethers.utils.recoverPublicKey(hash, sig));
+    const publicKey = `02${recoverPublicKey(digest, sig).slice(4, 68)}`;
```
Comment (ni3gavhane) [inline packages/claims/src/claims/claims.ts:65]: why is this '02' prefixed?

File: packages/did-ethr-resolver/src/implementations/operator.ts [hunk 2]
```diff
@@ -74,14 +78,15 @@ export class Operator extends Resolver implements IOperator {
...
       return this._keys.publicKey;
     }
-    const hash = await ethers.utils.keccak256(await this.getAddress());
-    const sig = await this._signer.signMessage(ethers.utils.arrayify(hash));
+    const hash = keccak256(await this.getAddress());
+    const sig = await this._signer.signMessage(arrayify(hash));
+    const digest = arrayify(hashMessage(arrayify(hash)));
     // eslint-disable-next-line no-return-assign
-    return this._keys.publicKey = ethers.utils.recoverPublicKey(hash, sig);
+    return this._keys.publicKey = `02${recoverPublicKey(digest, sig).slice(4, 68)}`;
```
Comment (ni3gavhane) [inline packages/did-ethr-resolver/src/implementations/operator.ts:89]: same here. 02?

File: packages/claims/src/claims/claims.ts (hunk 1 above)
Comment (JGiter) [reply packages/claims/src/claims/claims.ts:65]: Its prefix used in public keys. For uncompressed key the prefix is 04, for compressed are - 02, 03

File: packages/did-ethr-resolver/src/implementations/operator.ts (hunk 2 above)
Comment (JGiter) [reply packages/did-ethr-resolver/src/implementations/operator.ts:89]: Prefix used to differentiate between compressed and uncompressed key

**Review by ni3gavhane — APPROVED** [review]

**Pull request merged**